File: TermProject1/www/happiest_programmers.php
<?php

// Parse the sentiment file and create a dictionary with all the words and their sentiment values
$sentiment_file = fopen("data/AFINN-111.txt", "r");

$sentiment_values = array();

$total_sentiment_sum = 0;

# Read one line at a time
while (($line = fgets($sentiment_file)) !== false) {
	$term = preg_split('/\t/', $line);
	$sentiment_values[$term[0]] = $term[1];
}

// get contents of a file into a string
$filename = $_POST['tag_filename'];
$handle = fopen($filename, "r");
$contents = fread($handle, filesize($filename));

$json = json_decode($contents);

foreach ($json->items as $question) {
	# This holds the sum of the sentiment for this particular question
	$question_sentiment_sum = 0;
	
	# Make the tweet text all lower case in case there are some words with capitals, which wouldn't match our dict keys
	$body = strtolower($question->body);
	
	$body_terms = explode(" ", $body);
	
	# TODO: Remove all the punctuation from the beginning/end of the terms
	
	foreach ($body_terms as $term) {
		if (array_key_exists($term, $sentiment_values)) {
			$question_sentiment_sum += $sentiment_values[$term];
		}
	}
	
	$total_sentiment_sum += $question_sentiment_sum;
}

echo $total_sentiment_sum;

fclose($handle);

?>

File: TermProject1/src/happiest_programmers.py
'''
Created on Oct 24, 2013

@author: Ryan
'''

import sys
import json

# Parse the JSON file
if __name__ == '__main__':
    questions_file = open(sys.argv[1])
    sentiment_file = open(sys.argv[2])
    
    sentiment_scores = {} # initialize an empty dictionary
    
    for line in sentiment_file:
        term, score = line.split("\t") # The file is tab-delimited. "\t" means "tab character"
        sentiment_scores[term] = int(score)  # Convert the score to an integer.
    
    # This holds the sum of all the sentiments of every question in the file
    total_sentiment_sum = 0
    
    for line in questions_file:
        # Convert the line to a JSON object so we can process it
        questions = json.loads(line)
        
        for question in questions['items']:
            
            # Check if the tweet has the "text" key.
            if 'body' in question:
                # This holds the sum of the sentiment for this particular question
                question_sentiment_sum = 0
                
                question_text = question['body'].encode('utf-8')
                
                # Make the tweet text all lower case in case there are some words with capitals, which wouldn't match our dict keys
                question_text_lower = question_text.lower()
                
                terms = question_text_lower.split()
                
                # If any term ends with punctuation such as period, exclamation mark, question mark, then remove the punctuation.
                for i in range(len(terms)):
                    terms[i] = terms[i].rstrip('.!?,:;-')
                    
                for term in terms:
                    if term in sentiment_scores:
                        question_sentiment_sum += sentiment_scores[term]
                    
                #print '====== QUESTION TEXT =======\n' + question_text + '\n=======SENTIMENT VALUE========\n' + str(sentiment_sum)
                #break
            
                total_sentiment_sum += question_sentiment_sum
                #print question_sentiment_sum
        
        print total_sentiment_sum